### PR TITLE
Fix end 2 end pipeline

### PR DESCRIPTION
**Pull request by r00ta**
This PR aims to fix the errors in the e2e pipeline and remove the tests that are not testable (due to some missing features in python-libmaas)

### CODE REVIEW

**Review by SK1Y101 — APPROVED** [review]
Looks good, have you executed the tests with this new branch?

**Comment by r00ta** [comment]
> Looks good, have you executed the tests with this new branch?

I did locally :+1:

**Review by r00ta — COMMENTED** [review]
Explanation of the changes

File: terraform_test.py [hunk 1]
```diff
@@ -84,39 +84,31 @@ def check_maas_vlan(self, cfg: dict[str, Any]):
...
-        for key, value in cfg.items():
-            result = getattr(subnet, key)
-            assert value == result
 
     def check_maas_dns_domain(self, cfg: dict[str, Any]):
         domain = self._maas.domains.get(cfg["name"])
         assert domain is not None
 
-    def check_maas_dns_record(self, cfg: dict[str, Any]):
-        record = self._maas.dnsresources.get(cfg["fqdn"])
```
[the remark this answers is not in the document]
Comment (skatsaounis) [reply terraform_test.py:104]: I just created a PR to add the missing resources from the python-libmaas. Please take a look: https://github.com/maas/python-libmaas/pull/292

[the remark this answers is not in the document]
Comment (SK1Y101) [reply terraform_test.py:104]: this PR will probably merge before those changes are implemented & uploaded to PyPi, so it's worth adding an issue (or equivalent) to mark this change as temporary

File: terraform_test.tf [hunk 2]
```diff
@@ -90,26 +92,6 @@ resource "maas_dns_domain" "tf_test_domain" {
     authoritative = true
 }
 
-resource "maas_dns_record" "tf_test_record" {
-    type = "A/AAAA"
-    data = "10.10.10.1"
-    fqdn = "tftestrecord.${maas_dns_domain.tf_test_domain.name}"
-    depends_on = [maas_dns_domain.tf_test_domain]
-}
```
[the remark this answers is not in the document]
Comment (SK1Y101) [reply terraform_test.tf:98]: same point as above about the increasing libmaas coverage

**Review by SK1Y101 — APPROVED** [review]
New changes look good!

**Comment by r00ta** [comment]
Created https://warthogs.atlassian.net/browse/MAASENG-2177 , I will comment these tests instead of removing them

**Review by skatsaounis — APPROVED** [review]

**Pull request merged**